README.md

```diff
@@ -269,6 +269,16 @@ A `JsonApiClient::ResultSet` object should be paginatable with both `kaminari` a
 Person.where(name: 'Jeff').all
 ```
 
+## Headers
+
+You can specify the headers to send with all requests within a block with `with_headers`.
```

Comment:
Is this thread safe or does it modify a single connection object under the covers? If it's simply modifying the request that should be ok, but modifying anything with global state will cause auth token leakage in a multi-threaded environment.